localstack-core/localstack/services/sqs/provider.py

```diff
@@ -612,6 +614,34 @@ def check_fifo_id(fifo_id, parameter):
         )
 
 
+def get_sqs_protocol(request: Request) -> Literal["query", "json"]:
+    content_type = request.headers.get("Content-Type")
+    return "json" if content_type == "application/x-amz-json-1.0" else "query"
+
+
+def sqs_auto_protocol_aws_response_serializer(service_name: str, operation: str):
```

Comment:
this is a bit verbose and copies the logic of `aws_response_serializer` to get the `request` out of it, I'd like to make it cleaner but not sure how 🤔 there's no way of getting the proper protocol before we're inside the method itself.